python/ola/ClientWrapperTest.py

```diff
@@ -181,19 +182,32 @@ def d():
     self.start = datetime.datetime.now()
     wrapper.Run()
 
+    # Everything has been called
     self.assertIsNotNone(results.a_called)
     self.assertIsNotNone(results.b_called)
     self.assertIsNotNone(results.c_called)
     self.assertIsNotNone(results.d_called)
 
-    self.assertTrue(results.a_called - self.start <
-                    datetime.timedelta(milliseconds=5))
-    self.assertTrue(results.b_called - self.start >=
-                    datetime.timedelta(milliseconds=5))
-    self.assertTrue(results.c_called - self.start >=
-                    datetime.timedelta(milliseconds=10))
-    self.assertTrue(results.d_called - self.start >=
-                    datetime.timedelta(milliseconds=15))
+    # Check when the callbacks were called. Allow 500 microseconds of drift.
+    # Called immediately
+    a_diff = results.a_called - self.start
+    self.assertAlmostEqual(a_diff, datetime.timedelta(microseconds=0),
+                           delta=datetime.timedelta(microseconds=500))
```

Comment:
Personally I'd go for something like this, but I don't mind leaving a_diff if you'd rather.

```suggestion
    self.assertAlmostEqual((results.a_called - self.start), datetime.timedelta(microseconds=0),
                           delta=datetime.timedelta(microseconds=500))
```

Also have you done some dummy testing of this just to check assertAlmostEqual does actually work, e.g. change the offset of the B side a bit to confirm it passes (and breaks as appropriate)?

Reply:
Can confirm the test does what it says on the tin (the output when it fails is a bit verbose because of the `datetime.timedelta` everywhere).

I went with the local var mostly because I think it makes the test output more clear at a glance when it fails, like:

```
Traceback (most recent call last):
  File "./python/ola/ClientWrapperTest.py", line 194, in testEventLoop
    self.assertAlmostEqual(a_diff, datetime.timedelta(microseconds=0),
AssertionError: datetime.timedelta(microseconds=4) != datetime.timedelta(0) within datetime.timedelta(0) delta (datetime.timedelta(microseconds=4) difference)
```

Reply:
Fair enough, sounds good!